delphi-frontend/src/test/java/au/com/integradev/delphi/symbol/SymbolTableBuilderTest.java

```
@@ -174,4 +154,61 @@ void testStandardLibrarySearchPathShouldExcludeToolsUnits(
 
     assertThat(symbolTable.getUnitByPath(excludedPath.toString())).isNull();
   }
+
+  @Test
+  void testParsingErrorInImportShouldNotThrow(
+      @TempDir Path standardLibraryPath, @TempDir Path tempDir) throws IOException {
+    createStandardLibrary(standardLibraryPath);
+
+    Path includePath = tempDir.resolve("include");
+    Files.createDirectories(includePath);
+
+    Path errorUnitPath = includePath.resolve("ErrorUnit.pas");
+    Files.writeString(errorUnitPath, "unit ErrorUnit;");
+
+    Path sourceFilePath = tempDir.resolve("SourceFile.pas");
+    Files.writeString(
+        sourceFilePath,
+        "unit SourceFile;\n"
+            + "interface\n"
+            + "uses\n"
+            + "  ErrorUnit;"
+            + "implementation\n"
+            + "end.");
+
+    SymbolTableBuilder symbolTable =
+        SymbolTable.builder()
+            .preprocessorFactory(new DelphiPreprocessorFactory(Platform.WINDOWS))
+            .typeFactory(TypeFactoryUtils.defaultFactory())
+            .standardLibraryPath(standardLibraryPath)
+            .sourceFiles(List.of(sourceFilePath))
+            .searchPath(SearchPath.create(List.of(includePath)));
+
+    assertThatCode(symbolTable::build).doesNotThrowAnyException();
```

Maybe there should also be a test that checks if *lexing* errors are swallowed?